### PR TITLE
pythonPackages.quart: init at 0.15.1

### PR DESCRIPTION
**Pull request by dgliwka**
###### Motivation for this change
An asyncio reimplementation of the popular Flask microframework API

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/quart/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+, hypothesis
+, pytest
+, pytest-asyncio
+, pytest-cov
+, pytest-sugar
+}:
+
+buildPythonPackage rec {
+  pname = "quart";
+  version = "0.15.1";
```
Comment (fabaff) [inline pkgs/development/python-modules/quart/default.nix:21]:
```suggestion
  version = "0.15.1";
  disabled = pythonOlder "3.7";
```

see https://gitlab.com/pgjones/quart/-/blob/main/setup.py#L7

File: pkgs/development/python-modules/quart/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    pytest-asyncio
+    pytest-cov
+    pytest-sugar
+  ];
+
+  pythonImportsCheck = [ "quart" ];
+
+  meta = with lib; {
+    homepage = "https://pgjones.gitlab.io/quart/";
+    description = "An asyncio reimplementation of the popular Flask microframework API";
```
Comment (fabaff) [inline pkgs/development/python-modules/quart/default.nix:53]: I would suggest to not mention `flask`. Perhaps something like:

```suggestion
    description = "Python ASGI web microframework";
```

File: pkgs/development/python-modules/quart/default.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    hypercorn
+    itsdangerous
+    blinker
+    werkzeug
+  ];
+
+  checkInputs = [
+    mock
+    hypothesis
+    pytest
```
Comment (fabaff) [inline pkgs/development/python-modules/quart/default.nix:43]: Please use `pytestCheckHook`.

```suggestion
    pytestCheckHook
```

File: pkgs/development/python-modules/quart/default.nix [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    pytest-asyncio
+    pytest-cov
```
Comment (fabaff) [inline pkgs/development/python-modules/quart/default.nix:45]: Please patch the coverage part out. This doesn't add value for us.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129864` at 3d25a258 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.quart</li>
    <li>python39Packages.quart</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/quart/default.nix:53:0:
```
   |
53 |     description = "An asyncio reimplementation of the popular Flask microframework API";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/quart/default.nix (hunk 4 above)
Comment (dgliwka) [reply pkgs/development/python-modules/quart/default.nix]: Added patch to remove coverage part. Thanks for help :).

File: pkgs/development/python-modules/quart/default.nix [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitLab {
+    owner = "pgjones";
+    repo = pname;
+    rev = version;
+    sha256 = "036ip3lwpf4wkssmfpsdxnw46ncpafzwfswvsvkngbm9gnf1v4r7";
+  };
+
+  patches = [ ./no-coverage.patch ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/quart/default.nix:30]: Use substituteInPlace for this.

**Comment by gador** [comment]
Quart version 0.17 has been released. Also there are merge conflicts to be addressed.

File: pkgs/development/python-modules/quart/default.nix [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  propagatedBuildInputs = [
+    click
+    aiofiles
+    jinja2
+    hypercorn
+    itsdangerous
+    blinker
+    werkzeug
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/quart/default.nix:39]:
```suggestion
  ];
```

File: pkgs/development/python-modules/quart/default.nix [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+  checkInputs = [
+    pytestCheckHook
+    hypothesis
+    pytest-asyncio
+    pytest-cov
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/quart/default.nix:45]:
```suggestion
```
We don't want to generate cov.

File: pkgs/development/python-modules/quart/default.nix [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+, fetchFromGitLab
+, pythonOlder
+, click
+, aiofiles
+, werkzeug
+, jinja2
+, hypercorn
+, itsdangerous
+, blinker
+, hypothesis
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/quart/default.nix:12]:
```suggestion
, hypothesis
, mock
```

**Pull request closed without merging**